src/llmcompressor/transformers/sparsification/sparse_model.py

```diff
@@ -153,6 +138,14 @@ def skip(*args, **kwargs):
 
         return model
 
+    # Add the from_pretrained method to the new class
+    setattr(new_class, 'from_pretrained', from_pretrained)
+
+    return new_class
+
+
+SparseAutoModelForCausalLM = wrap_hf_model_class(AutoModelForCausalLM)
```

Comment:
This definition should be moved to a separate file to reduce import times

Reply:
It is the same as before so I'm not worried about it

Reply:
Agree!